pkg/cvo/sync_worker.go

```diff
@@ -100,7 +101,8 @@ type CapabilityStatus struct {
 type SyncWorkerStatus struct {
 	Generation int64
 
-	Failure error
+	FailureSummary error
+	Failures       []error
```

Comment:
I think we can avoid having to pass these in parallel.  I've floated #1041 with a sketch showing `Unwrap()` and `Errors()` to walk the nested errors inside `Failure`.  It probably needs more filling out with the structured information we store for each node in the tree, but it will avoid us having to worry about things like "what if  the actor populating `SyncWorkerStatus` puts something in `FailureSummary` without updating `Failures` to hold the appropriate children?".